### ISSUE TITLE
April Update Cleanup

### ISSUE THREAD

**Issue by iBotPeaches** [TrackerNetwork/DestinyStatus#37]
1. New PoE (Taken) is not like the others in terms of API. Each week has a different "activityHash", but then there is a Hash for the Playlist. This is like the new Nightfall endpoint. We have to change up the detection because in the current state it will never match thus always return 0 completed and never completed for that week. Additional there are Skulls and Bonuses instead of flat out Skulls. Finally there are also scores.
2. The advisors endpoint might be easier to use than levering activities and summary. I could be spitting nonsense, but I just need to familiarize myself with the endpoints and how they relate to DS. Maybe we can cut down # of requests made.
3. The max light & level is wrong in code but doesn't seem to matter.
4. Could clean-up homepage. Upcoming Trials card, Weekly/Daily Crucible cards, Arena Taken PoE card.

Notes as I discovered changed things. Nothing dead and broken so thats good.

**Comment by sgtfrankieboy** [comment]
I don't think we can cut our # requests for advisors as we already use 1 for that, we just have to change it over to another. Because the new `activities` property in the advisor doesn't exists under the `destiny/advisors/` endpoint which we currently use. So we will need to change that all up to use a user specific advisor endpoint. Same goes with `elderChallenge`.

**Comment by bladefist** [comment]
@iBotPeaches Any help you can provide would be great.  I have a lot of updates to make to DTR.  We may need to put a banner up on the homepage of DS to ask for more community support, unless you think you can take this on.

**Comment by iBotPeaches** [comment]
I did some reading and not sure how this `/advisor` endpoint helps. I look at the information it provides and we'd still need to iterate for each character. I was under the improper impression that this changed endpoint was all characters in one endpoint.

Number 1 in original report looks like the big change. Bringing in the tracking and showing of PoE Taken. I wrote the same code for the Nightfall so I can easily adapt that for PoE Taken edition. A bit fuzzy on how to do the "completions" for it though since each week will have a different hash till it rotates back, will have to peek at the `aggregate` endpoint again.

I kinda put Destiny aside and went to Halo5, so a bit out of it, but played a few hours last night and noticed there were scores for the taken PoE. I'm not sure if this is tracked/recorded somewhere, but if so I'll add that into the checklist tab.

I'll get a PR in soon that fixes 1 & 3. I'll float a different PR for number 4 if I can figure out a cool way to display the Arena Taken one with bonus and regular skulls.

TLDR - I'll take a shot at it

**Comment by bladefist** [comment]
@iBotPeaches  Thanks so much.  Ya I played first time in a while too.  What do you mean scores for PoE? I guess I don't remember seeing that ingame. you think it's important?

**Comment by iBotPeaches** [comment]
I just did some random LFG and we played 3 Taken PoE (Lvl42) and had to get our "score" above 30,000 each time. That "bonus" point skull for precision kills gave us bonus points. Then we unlocked a cumulative 90,000 point thingy and per-game 30,000 thingy. They were attached to some item that when unlocked gave us gear.

Thing is, after we hit 90,000 and played a 4th. It logged it. So my card said 123,000/90,000. Unfortunately typing this out I just realized I couldn't view other players scores so maybe not visible to us.

**Comment by bladefist** [comment]
Ah, I haven't been able to play that quite yet.  You're probably right but it's worth confirming.

**Comment by sgtfrankieboy** [comment]
I believe the API endpoint `platform/destiny/{membershipType}/account/{membershipId}/{characterId}/advisors` provides this information under the property `elderChallenge`. However that property and all others has been marked as depreciated because of the new `activities` property that has information on all the advisors in a more structured way. But the new `advisors` property is missing some information like the players progression which does exists in the `elderChallenge` property. [Example Request](http://pastebin.com/raw/H0c5cmDy). 

To show the elder challenge on the DestinyStatus homepage we will need to use that endpoint instead of the public one ('platform/destiny/advisors') because it doesn't have that data.

[Thread about new additions on Bungie.net](https://www.bungie.net/en/Clan/Post/39966/199462093/0/0)

**Comment by iBotPeaches** [comment]
Okay, hit some weird bug or misunderstanding. Writing it out to maybe help when I return to it. So each Arena has an `activityHash`. I believe in the current state of DestinyStatus if you complete the Level34 PoE and its on the week of "Machine Wrath" that completion won't count on the week that the Level34 "Forever Eater" iteration of Level34 appears. Or in the short, the completions is not "ALL" the Level34 PoE completed, its just that `activityHash`.

So every time you view the profile, you are only get analytics of the week currently active. This makes sense in a bit for the original PoE's since the name changed. IE - This week on the Level34 PoE, I have 1 completion on my titan. Whatever next week's Level34 PoE will change that and thus my completions.

Shifting gears to the new Taken PoE. The name is "Challenge of the Elders". I expect next week to be "Challenge of the Elders" except with 3 new mini bosses. Right now we are on `1673931290` which is Overmind (Vex), Sylok (Hive) & Pilot (Fallen). So I thought I would be smart and record the `Type` of these activities which `3705723572` is [1] below. Then I could tabulate all the PoE under that Type (The Level42 challenge modes) and count completions. However, PoE from the original iteration (Level32, Level34, Level35) have this same `activityTypeHash` too which makes counting pretty hard. So I'll have to rely on probably `Level` which is 42 and thus smaller in other PoEs.

[1]

```
<?php return array (
  'activityTypeHash' => 3705723572,
  'identifier' => 'ARENA_CHALLENGE',
  'activityTypeName' => 'Challenge of the Elders',
  'icon' => '/common/destiny_content/icons/84c1d7eefdea308020752710d43f823f.png',
  'activeBackgroundVirtualPath' => '/img/destiny_content/ActivityTypeBackground/Active/ministory.png',
  'completedBackgroundVirtualPath' => '/img/destiny_content/ActivityTypeBackground/Completed/ministory.png',
  'hiddenOverrideVirtualPath' => '/img/destiny_content/ActivityTypeBackground/Hidden/ministory.png',
  'tooltipBackgroundVirtualPath' => '/img/destiny_content/ActivityTypeBackground/Tooltip/ministory.png',
  'enlargedActiveBackgroundVirtualPath' => '/img/destiny_content/ActivityTypeBackground/Active/ministory.png',
  'enlargedCompletedBackgroundVirtualPath' => '/img/destiny_content/ActivityTypeBackground/Completed/ministory.png',
  'enlargedHiddenOverrideVirtualPath' => '/img/destiny_content/ActivityTypeBackground/Hidden/ministory.png',
  'enlargedTooltipBackgroundVirtualPath' => '/img/destiny_content/ActivityTypeBackground/Tooltip/ministory.png',
  'order' => 1,
  'hash' => 3705723572,
  'index' => 0,
);
```

**Comment by iBotPeaches** [comment]
> Oh, I almost forgot - the new "Challenge of the Elders" mode will have its own Activity Mode - "arenaElderChallenge" - but it is a known issue that while it has its own Activity Mode, **it shares the same Activity Type as the old "Prison of Elders Challenge Mode" activities**... so if you're doing custom filtering, be sure to filter on activity mode. The old activity type has been, unfortunately, renamed to "Challenge of the Elders" mode - we're looking into a long term fix, but for now that Activity Type has a very misleading name, as the type itself can refer to old POE Challenge Mode activities as well. (thanks Obama)

Doh. Should have read the group first. 

Progress (Just need working completions) 

![test](https://cloud.githubusercontent.com/assets/611784/14673712/9360401a-06cd-11e6-8584-ea17bf6861af.png)

The public advisors call seems weak compared to the authenticated one, so not possible to get the info I need for PoE Challenge on Homepage.

**Comment by sgtfrankieboy** [comment]
It is no longer needed to authenticate for the private advisor endpoint. They've made all the data public unless the user explicitly turned it off.

**Comment by iBotPeaches** [comment]
Ah sweet. So can use a dummy account solely to use that endpoint to populate the homepage with data then.

**Comment by macnode** [comment]
I play A LOT and do all the possible activities every week, and know how do calls to the Bungie API. Feel free to make calls against my account (XB1: L0r3) or let me know if you need anything. Ill post a minor bug I found.

**Comment by TRNIntegrations** [comment]
@macnode thanks!

**Comment by macnode** [comment]
@TRNIntegrations Do you happen to know if raid completions are being calculated via Activity History and looping though the history (yuck)? Thats the only way I know how right now.

http://www.bungie.net/Platform/Destiny/Stats/ActivityHistory/[platform]/[account]/[charcter]

Or is Bungie providing a better way?

**Comment by TRNIntegrations** [comment]
They are but there is a new way to do it that doesn't require any yuckyness.  It's the new character advisors api point.  Are you able to look into? My php skills are lacking

**Comment by iBotPeaches** [comment]
I believe right now completions are calculated via the aggregate endpoint - https://github.com/TrackerNetwork/DestinyStatus/blob/master/app/destiny/DestinyPlatform.php#L125

The looping through history (while yucky) is how we determine if that activity has been completed this reset or not (the yellow check). This I believe is possible to obtain from the new Advisors endpoint. I've slacked on switching to this and/or understanding everything since in the past it lacked some data that would allow a complete switch to it vs just adding more requests.

**Comment by macnode** [comment]
If you just want raid completions as a whole, there is grim card '603010'. It will only give you the total completion number. Not broken down by which raid.

https://www.bungie.net/Platform/Destiny/Vanguard/Grimoire/1/$memID/?single=603010

**Comment by iBotPeaches** [comment]
The original report here to add TTK PoE was fixed during the move to Laravel 5.1 when I switched to the advisors/v2 endpoint.
